crates/steel-repl/src/repl.rs

```diff
@@ -97,31 +97,37 @@ fn finish_load_or_interrupt(vm: &mut Engine, exprs: String, path: PathBuf) {
     }
 }
 
-fn finish_or_interrupt(vm: &mut Engine, line: String, print_time: bool) {
-    let now = Instant::now();
+fn finish_or_interrupt(vm: &mut Engine, line: String) {
+    let values = match vm.compile_and_run_raw_program(line) {
+        Ok(values) => values,
+        Err(error) => {
+            vm.raise_error(error);
 
-    let res = vm.compile_and_run_raw_program(line);
+            return;
+        }
+    };
 
-    match res {
-        Ok(r) => r.into_iter().for_each(|x| match x {
+    let len = values.len();
+
+    for (i, value) in values.into_iter().enumerate() {
+        let last = i == len - 1;
+
+        if last {
+            vm.register_value("$1", value.clone());
```

Comment:
Ideally, a `(define $1 "foo")` would shadow this "magic" binding forever, but I imagine that would require deeper changes to the VM/engine.

Reply:
That should be the behavior as is with `register_value`, so we're good here

Reply:
FWIW I meant more like:

```scheme
> "foobar"
"foobar"
> $1
=> "foobar"
> (define $1 'xyz)
> "potato"
=> "potato"
> $1  ;; after a explicit `define`, the "magic binding" is gone
=> 'xyz
```

Reply:
I see, let me take a look and evaluate the difficulty of adding that